cloud-sql/mysql/sqlalchemy/main.py

```diff
@@ -62,17 +44,87 @@
     # 'pool_timeout' is the maximum number of seconds to wait when retrieving a
     # new connection from the pool. After the specified amount of time, an
     # exception will be thrown.
-    pool_timeout=30,  # 30 seconds
+    "pool_timeout": 30,  # 30 seconds
     # [END cloud_sql_mysql_sqlalchemy_timeout]
     # [START cloud_sql_mysql_sqlalchemy_lifetime]
     # 'pool_recycle' is the maximum number of seconds a connection can persist.
     # Connections that live longer than the specified amount of time will be
     # reestablished
-    pool_recycle=1800,  # 30 minutes
+    "pool_recycle": 1800,  # 30 minutes
     # [END cloud_sql_mysql_sqlalchemy_lifetime]
-    # [END_EXCLUDE]
-)
-# [END cloud_sql_mysql_sqlalchemy_create]
+}
+
+app = Flask(__name__)
+
+logger = logging.getLogger()
+
+
+def init_connection_engine():
+    if os.environ.get("DB_HOST"):
+        return init_tcp_connection_engine()
+    else:
+        return init_unix_connection_engine()
+
+
+def init_tcp_connection_engine():
+    # [START cloud_sql_mysql_sqlalchemy_create_tcp]
+    db_socket_addr = os.environ.get("DB_HOST").split(":")
+
+    # Extract host and port from socket address
+    db_host = db_socket_addr[0]
+    db_port = int(db_socket_addr[1])
+
+    return sqlalchemy.create_engine(
+        # Equivalent URL:
+        # mysql+pymysql://<db_user>:<db_pass>@<db_host>:<db_port>/<db_name>
+        sqlalchemy.engine.url.URL(
+            drivername="mysql+pymysql",
+            username=db_user,
+            password=db_pass,
+            host=db_host,
+            port=db_port,
+            database=db_name,
+        ),
+        # ... Specify additional properties here.
+        # [START_EXCLUDE]
+        **db_config
```

Comment:
Question for repo owners: are we still requiring 2.7 compatibility for all tests or is this fine?